### ISSUE TITLE
Possible regression of has_many :through in 5.2.0.rc1

### ISSUE THREAD

**Issue by repinel** [rails/rails#32008]
This is a possible regression from `5.1.4` to `5.2.0.rc1`. It seems to have been caused by 3acc5d6.

### Steps to reproduce

```
# frozen_string_literal: true

begin
  require "bundler/inline"
rescue LoadError => e
  $stderr.puts "Bundler version 1.10 or later is required. Please update your Bundler"
  raise e
end

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "activerecord", "5.2.0.rc1" # works with "5.1.4"
  gem "sqlite3"
end

require "active_record"
require "minitest/autorun"
require "logger"

# Ensure backward compatibility with Minitest 4
Minitest::Test = MiniTest::Unit::TestCase unless defined?(Minitest::Test)

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
  end

  create_table :comments, force: true do |t|
    t.integer :post_id
    t.string :content
  end

  create_table :tags, force: true do |t|
    t.integer :comment_id
    t.string :name
  end
end

class Post < ActiveRecord::Base
  has_many :comments, -> { order :content }
  has_many :tags, -> { order :name }, through: :comments
end

class Comment < ActiveRecord::Base
  belongs_to :post
  has_many :tags
end

class Tag < ActiveRecord::Base
  belongs_to :comment
end

class BugTest < Minitest::Test
  def test_association_stuff
    post = Post.create!
    comment2 = post.comments.create! content: "bad"
    comment1 = post.comments.create! content: "awesome"
    comment3 = post.comments.create! content: "good"

    assert_equal [comment1, comment2, comment3], post.reload.comments.to_a

    tag3 = comment1.tags.create! name: "c"
    tag2 = comment1.tags.create! name: "b"
    tag1 = comment2.tags.create! name: "a"

    assert_equal [tag1, tag2, tag3], post.reload.tags.to_a
  end
end
```

### Expected behavior

`SELECT "tags".* FROM "tags" INNER JOIN "comments" ON "tags"."comment_id" = "comments"."id" WHERE "comments"."post_id" = ? ORDER BY "tags"."name" ASC, "comments"."content" ASC`

### Actual behavior

`SELECT "tags".* FROM "tags" INNER JOIN "comments" ON "tags"."comment_id" = "comments"."id" WHERE "comments"."post_id" = ? ORDER BY "comments"."content" ASC, "tags"."name" ASC`

### System configuration
**Rails version**: 5.2.0.rc1

**Ruby version**: 2.3.0

**Comment by repinel** [comment]
Exactly, I wasn't sure if it was intentional or not. A CHANGELOG entry could help.

**Comment by kamipo** [comment]
Thanks for the report.
Changing the order of scope evaluation is required to fix unscoping through scope, but we can keep the sorting order as before. I've fixed in 5c6d6fd30af2208a86f97ce0ca303c6c59700246.
